### ISSUE TITLE
Using a MixedCase nodepool name prevents cluster creation

### ISSUE THREAD

**Issue by wallrj** [jetstack/navigator#94]
Using a MixedCase nodepool name results in the following warning event and results in a partial cluster.

```
kubectl get events --namespace navigator-e2e-database1
LAST SEEN   FIRST SEEN   COUNT     NAME                    KIND                   SUBOBJECT   TYPE      REASON    SOURCE                 MESSAGE
3m          3m           1         demo.14f3446201bd7c7f   ElasticsearchCluster               Warning   ErrSync   navigator-controller   Error syncing config map: error ensuring "demo"/"MasterPool" nodepool configmap: ConfigMap "es-demo-MasterPool" is invalid: metadata.name: Invalid value: "es-demo-MasterPool": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```
/kind bug

Ideally there'd be validation when the ElasticSearchCluster configuration is created.
Admission controller?

**Comment by munnerz** [comment]
Yep - we need an admission controller to perform validation here.

**Comment by munnerz** [comment]
Closed by #199 - it's now no longer possible to have caps in a node pool name

/close
